packages/args.ol

```diff
@@ -0,0 +1,115 @@
+from string-utils import StringUtils
+from console import Console
+
+
+type GetParsedArgsRequest {
+    args*: string 
+    argsMap*: void {
+        name: string        // name of the expected argument
+        optional: bool      // specifies if the argument is optional
+        description: string // describes the argument
+        isFlag: bool       // if true, the argument is a flag (e.g. --verbose)
+    }
+
+}
+
+type GetParsedResponse {
+    parsedArgs* {
+        name: string 
+        value: string | void
+    }
+}
+
+interface ArgsInterface {
+    RequestResponse:
+        getParsedArgs( GetParsedArgsRequest )( GetParsedResponse ) throws ParametersError( string ) Help
```

Comment:
Why is Help a fault? Are we forcing everyone to use --help as escape hatch? I'd just return the response normally and then leave it to the client to check if --help was specified, it looks like non-exceptional logic.

Reply:
Because args has been thouth for being used from console. Usually when you type --help the command prints out the help list and then exit.